AmplifyPlugins/DataStore/Sources/AWSDataStorePlugin/Core/DataStoreModelDecoder.swift

```diff
@@ -15,18 +15,33 @@ public struct DataStoreModelDecoder: ModelProviderDecoder {
     
     /// Metadata that contains the foreign key value of a parent model, which is the primary key of the model to be loaded.
     struct Metadata: Codable {
-        let identifier: String?
+        let identifiers: [LazyReferenceIdentifier]
```

Comment:
maybe this can just be an array of strings, `[String]`, then we don't have to pad it from Statement+Model. When DataStoreModelProvider.init(metadata) is called, it can create the LazyReferenceIdentifier's with the following outputs:

```
// regular
LazyReferenceIdentifier(name: "id", value:"id123")

// CPK
[LazyReferenceIdentifier(name: "@@primaryKey", value:"id123"),
LazyReferenceIdentifier(name: "", value:"title123")] 
```

Then we can change the `load()` slightly to always use the first name ("id" or "@@primaryKey") and concat the values.